docs/jax.lax.rst

```diff
@@ -162,6 +162,14 @@ Custom gradient operators
     custom_linear_solve
     custom_root
 
+Function operators
```

Comment:
Is `named_call` an operator (I understand that the symbol is a function that operates on functions, but since it is a higher order function I find the word "operator" a bit confusing, might just be me..)?

If we keep this in `jax.lax` would put this at the bottom of the file under a heading `Debugging` or `Compilation`. Alternatively we could move this to `jax.named_call` and then document it under `Just-in-time-compilation (jit)`. I think that might actually make the most sense since `named_call` is only useful once JAX programs are staged out.

@gnecula wdyt?

Reply:
I like the idea of putting this in `jax` instead of `lax`.

Reply:
Done. I created a `named_call.py` under `jax/_src` which has the implementation, and another `jax/named_call.py` that only contains only imports. Then I exposed both symbols directly under jax (`jax.named_call` and `jax.named_call_p` )